editor/src/communication/broadcast_message.rs

```diff
@@ -0,0 +1,32 @@
+use crate::message_prelude::*;
+
+use serde::{Deserialize, Serialize};
+
+#[impl_message(Message, Broadcast)]
+#[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub enum BroadcastMessage {
+	SubscribeSignal { on: BroadcastSignal, send: Box<Message> },
+	UnsubscribeSignal { on: BroadcastSignal, message: Box<Message> },
+	TriggerSignal { signal: BroadcastSignal },
+	TriggerSignalFront { signal: BroadcastSignal },
```

Comment:
Max and I discussed this, I advocated for avoidance of the word "immediate" because it implies that it happens, well, immediately. But that isn't true: it puts it on the front of the queue and then perhaps others in this, or a function elsewhere in the callstack, may enqueue more stuff at the front before it finally returns all the way back to the dispatcher and then executes them in order of the queue (possibly triggering others to insert more messages first in the queue also). So it really isn't immediate.

Using that word also reinforces the misconception, which was already a point of confusion for past contributor, that a message added to the queue happens right away rather than waiting until the entire function (and any ancestor functions in the call stack) return, because it works by mutating the queue and eventually returning back to the dispatcher rather than working in the more traditional manner of an immediate function call.

"Delayed" is also confusing since it seems to imply some time delay, rather than being put at the back of the message queue. I'm certainly open to more naming suggestions but I think this is a pretty reasonable one that I'm happy with.

Reply:
The second point still stands though, I think we only need one of the messages anyway, making the naming discussion obsolete

Reply:
Ah, you're right, thanks for pointing that out. I understand that you mean now.